// storybook-readme/const.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LAYOUT_TYPE_PROPS_TABLE = exports.LAYOUT_TYPE_STORY_SOURCE = exports.LAYOUT_TYPE_STORY = exports.LAYOUT_TYPE_HEADER_MD = exports.LAYOUT_TYPE_FOOTER_MD = exports.LAYOUT_TYPE_MD = exports.MARKER_HIDDEN = exports.MARKER_PROPS_TABLE = exports.MARKER_STORY_SOURCE = exports.MARKER_STORY = exports.CHANNEL_SET_SIDEBAR_DOCS = void 0;
var CHANNEL_SET_SIDEBAR_DOCS = 'SET_SIDEPAGE_DOCS';
exports.CHANNEL_SET_SIDEBAR_DOCS = CHANNEL_SET_SIDEBAR_DOCS;
var MARKER_STORY = '<!-- STORY -->';
exports.MARKER_STORY = MARKER_STORY;
var MARKER_STORY_SOURCE = '<!-- SOURCE -->';
exports.MARKER_STORY_SOURCE = MARKER_STORY_SOURCE;
var MARKER_PROPS_TABLE = '<!-- PROPS -->';
exports.MARKER_PROPS_TABLE = MARKER_PROPS_TABLE;
var MARKER_HIDDEN = /<!-- STORY HIDE START -->(.|\s)*?<!-- STORY HIDE END -->/g;
exports.MARKER_HIDDEN = MARKER_HIDDEN;
var LAYOUT_TYPE_MD = 'MD';
exports.LAYOUT_TYPE_MD = LAYOUT_TYPE_MD;
var LAYOUT_TYPE_FOOTER_MD = 'FOOTER_MD';
exports.LAYOUT_TYPE_FOOTER_MD = LAYOUT_TYPE_FOOTER_MD;
var LAYOUT_TYPE_HEADER_MD = 'HEADER_MD';
exports.LAYOUT_TYPE_HEADER_MD = LAYOUT_TYPE_HEADER_MD;
var LAYOUT_TYPE_STORY = 'STORY';
exports.LAYOUT_TYPE_STORY = LAYOUT_TYPE_STORY;
var LAYOUT_TYPE_STORY_SOURCE = 'STORY_SOURCE';
exports.LAYOUT_TYPE_STORY_SOURCE = LAYOUT_TYPE_STORY_SOURCE;
var LAYOUT_TYPE_PROPS_TABLE = 'PROPS';
exports.LAYOUT_TYPE_PROPS_TABLE = LAYOUT_TYPE_PROPS_TABLE;
