// storybook-readme/registerWithPanelTitle.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = registerWithPanelTitle;

var _react = _interopRequireDefault(require("react"));

var _addons = _interopRequireDefault(require("@storybook/addons"));

var _ReadmeSidebar = _interopRequireDefault(require("./components/ReadmeSidebar"));

var ADDON_ID = 'REACT_STORYBOOK/readme';
var PANEL_NAME = 'REACT_STORYBOOK/readme/panel';
var PANEL_TITLE = 'Readme';

function registerWithPanelTitle() {
  var name = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : PANEL_TITLE;

  _addons.default.register(ADDON_ID, function (api) {
    var channel = _addons.default.getChannel();

    _addons.default.addPanel(PANEL_NAME, {
      title: name,
      render: function render(props) {
        // compatibility with storybook@3.x
        var active = !props || props.active;
        return _react.default.createElement(_ReadmeSidebar.default, {
          key: 'ReadmeSidebar',
          api: api,
          active: active,
          channel: channel,
          onStory: api.onStory
        });
      }
    });
  });
}
