// storybook-readme/html/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.addReadme = void 0;

var _addons = _interopRequireWildcard(require("@storybook/addons"));

var _getDocsLayout = _interopRequireDefault(require("../services/getDocsLayout"));

var _getParameters = _interopRequireDefault(require("../services/getParameters"));

var _const = require("../const");

var addReadme = (0, _addons.makeDecorator)({
  name: 'addReadme',
  parameterName: 'readme',
  wrapper: function wrapper(getStory, context) {
    var parameters = (0, _getParameters.default)(context);
    var story = getStory(context);
    var layout = parameters.layout ? parameters.layout : (0, _getDocsLayout.default)({
      footer: parameters.footer || '',
      header: parameters.header || '',
      md: parameters.content || '',
      excludePropTables: parameters.excludePropTables || [],
      includePropTables: parameters.includePropTables || [],
      story: story
    });

    var channel = _addons.default.getChannel();

    if (parameters.sidebar) {
      var sidebarLayout = (0, _getDocsLayout.default)({
        footer: parameters.footer || '',
        header: parameters.header || '',
        md: parameters.sidebar,
        excludePropTables: parameters.excludePropTables || [],
        includePropTables: parameters.includePropTables || [],
        story: story
      });
      channel.emit(_const.CHANNEL_SET_SIDEBAR_DOCS, {
        layout: sidebarLayout,
        theme: parameters.theme,
        codeTheme: parameters.codeTheme
      });
    }

    return story;
  }
});
exports.addReadme = addReadme;
