// storybook-readme/services/config.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.addFooter = addFooter;
exports.addHeader = addHeader;
exports.addStoryPreview = addStoryPreview;
exports.addDocPreview = addDocPreview;
exports.addHeaderPreview = addHeaderPreview;
exports.addFooterPreview = addFooterPreview;
exports.getConfig = getConfig;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var config = {};

function addFooter(footer) {
  config.footer = footer;
}

function addHeader(header) {
  config.header = header;
}

function addStoryPreview(Preview) {
  config.StoryPreview = Preview;
}

function addDocPreview(Preview) {
  config.DocPreview = Preview;
}

function addHeaderPreview(Preview) {
  config.HeaderPreview = Preview;
}

function addFooterPreview(Preview) {
  config.FooterPreview = Preview;
}

function getConfig() {
  return (0, _objectSpread2.default)({}, config);
}
