// storybook-readme/backwardCompatibility/env/react/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ReadmeContent = _interopRequireDefault(require("../../../components/ReadmeContent"));

var _common = _interopRequireDefault(require("../common"));

var _const = require("../../../const");

var _getDocsLayout = _interopRequireDefault(require("../../../services/getDocsLayout"));

function withDocsCallAsHoc(_ref) {
  var md = _ref.md,
      story = _ref.story;
  return function (context) {
    var layout = (0, _getDocsLayout.default)({
      md: md,
      story: _react.default.createElement(_react.default.Fragment, null, story(context))
    });
    return _react.default.createElement(_ReadmeContent.default, {
      layout: layout,
      types: [_const.LAYOUT_TYPE_PROPS_TABLE, _const.LAYOUT_TYPE_STORY, _const.LAYOUT_TYPE_MD]
    });
  };
}

function withDocsCallAsDecorator(_ref2) {
  var md = _ref2.md;
  return function (story, context) {
    var layout = (0, _getDocsLayout.default)({
      md: md,
      story: _react.default.createElement(_react.default.Fragment, null, story(context))
    });
    return _react.default.createElement(_ReadmeContent.default, {
      layout: layout,
      types: [_const.LAYOUT_TYPE_PROPS_TABLE, _const.LAYOUT_TYPE_STORY, _const.LAYOUT_TYPE_MD]
    });
  };
}

function doc(_ref3) {
  var md = _ref3.md;
  var layout = (0, _getDocsLayout.default)({
    md: md,
    story: null
  });
  return function () {
    return _react.default.createElement(_ReadmeContent.default, {
      layout: layout,
      types: [_const.LAYOUT_TYPE_PROPS_TABLE, _const.LAYOUT_TYPE_STORY, _const.LAYOUT_TYPE_MD]
    });
  };
}

var _default = {
  doc: doc,
  withReadme: _common.default.withReadme,
  withDocs: {
    callAsDecorator: withDocsCallAsDecorator,
    callAsHoc: withDocsCallAsHoc
  }
};
exports.default = _default;
