// storybook-readme/components/Marked/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _marked = _interopRequireDefault(require("../../services/marked"));

var _highlite = _interopRequireDefault(require("../../services/highlite"));

var _transformEmojis = _interopRequireDefault(require("../../services/transformEmojis"));

var Marked =
/*#__PURE__*/
function (_React$Component) {
  (0, _inherits2.default)(Marked, _React$Component);

  function Marked() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, Marked);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(Marked)).call.apply(_getPrototypeOf2, [this].concat(args)));
    _this.ref = null;

    _this.handleRef = function (ref) {
      _this.ref = ref;

      if (_this.ref) {
        (0, _highlite.default)(_this.ref);
      }
    };

    return _this;
  }

  (0, _createClass2.default)(Marked, [{
    key: "componentDidUpdate",
    value: function componentDidUpdate() {
      if (this.ref) {
        (0, _highlite.default)(this.ref);
      }
    }
  }, {
    key: "render",
    value: function render() {
      var md = this.props.md;
      return _react.default.createElement("div", {
        ref: this.handleRef,
        className: 'markdown-body',
        dangerouslySetInnerHTML: {
          __html: (0, _marked.default)((0, _transformEmojis.default)(md))
        }
      });
    }
  }]);
  return Marked;
}(_react.default.Component);

exports.default = Marked;
