// storybook-readme/backwardCompatibility/env/vue/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _common = _interopRequireDefault(require("../common"));

var _ReadmeContent = _interopRequireDefault(require("../../../vue/components/ReadmeContent"));

var _getDocsLayout = _interopRequireDefault(require("../../../services/getDocsLayout"));

var _getParameters = _interopRequireDefault(require("../../../services/getParameters"));

var _const = require("../../../const");

function withDocsCallAsHoc(_ref) {
  var md = _ref.md,
      story = _ref.story;
  return function (context) {
    var layout = (0, _getDocsLayout.default)({
      md: md,
      story: story(context)
    });
    var parameters = (0, _getParameters.default)(context);
    return {
      data: function data() {
        return {
          parameters: parameters,
          layout: layout,
          types: [_const.LAYOUT_TYPE_PROPS_TABLE, _const.LAYOUT_TYPE_STORY, _const.LAYOUT_TYPE_MD]
        };
      },
      components: {
        'readme-content': _ReadmeContent.default
      },
      template: "<readme-content \n        v-bind:backward=\"true\"\n        v-bind:types=\"types\"\n        v-bind:withPreview=\"true\"\n        v-bind:theme=\"parameters.theme\"\n        v-bind:codeTheme=\"parameters.codeTheme\"\n        v-bind:layout=\"layout\" />"
    };
  };
}

function withDocsCallAsDecorator(_ref2) {
  var md = _ref2.md;
  return function (story, context) {
    var layout = (0, _getDocsLayout.default)({
      md: md,
      story: story(context)
    });
    var parameters = (0, _getParameters.default)(context);
    return {
      data: function data() {
        return {
          parameters: parameters,
          layout: layout,
          types: [_const.LAYOUT_TYPE_PROPS_TABLE, _const.LAYOUT_TYPE_STORY, _const.LAYOUT_TYPE_MD]
        };
      },
      components: {
        'readme-content': _ReadmeContent.default
      },
      template: "<readme-content\n        v-bind:backward=\"true\"\n        v-bind:types=\"types\"\n        v-bind:withPreview=\"true\"\n        v-bind:theme=\"parameters.theme\"\n        v-bind:codeTheme=\"parameters.codeTheme\"\n        v-bind:layout=\"layout\" />"
    };
  };
}

function doc(_ref3) {
  var md = _ref3.md;
  var layout = (0, _getDocsLayout.default)({
    md: md,
    story: null
  });
  return function (context) {
    var parameters = (0, _getParameters.default)(context);
    return {
      data: function data() {
        return {
          parameters: parameters,
          layout: layout
        };
      },
      components: {
        'readme-content': _ReadmeContent.default
      },
      template: "<readme-content\n        v-bind:backward=\"true\" \n        v-bind:theme=\"parameters.theme\"\n        v-bind:codeTheme=\"parameters.codeTheme\"\n        v-bind:layout=\"layout\" />"
    };
  };
}

var _default = {
  doc: doc,
  withReadme: _common.default.withReadme,
  withDocs: {
    callAsDecorator: withDocsCallAsDecorator,
    callAsHoc: withDocsCallAsHoc
  }
};
exports.default = _default;
