// storybook-readme/backwardCompatibility/env/common/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _addons = _interopRequireDefault(require("@storybook/addons"));

var _const = require("../../../const");

var _getDocsLayout = _interopRequireDefault(require("../../../services/getDocsLayout"));

var _getParameters = _interopRequireDefault(require("../../../services/getParameters"));

function withReadmeCallAsDecorator(_ref) {
  var md = _ref.md;
  return function (story, context) {
    var storyComponent = story(context);
    var layout = (0, _getDocsLayout.default)({
      md: md,
      story: storyComponent
    });
    var parameters = (0, _getParameters.default)(context);

    _addons.default.getChannel().emit(_const.CHANNEL_SET_SIDEBAR_DOCS, {
      layout: layout,
      theme: parameters.theme,
      codeTheme: parameters.codeTheme
    });

    return storyComponent;
  };
}

function withReadmeCallAsHoc(_ref2) {
  var md = _ref2.md,
      story = _ref2.story;
  return function (context) {
    var storyComponent = story(context);
    var layout = (0, _getDocsLayout.default)({
      md: md,
      story: storyComponent
    });
    var parameters = (0, _getParameters.default)(context);

    _addons.default.getChannel().emit(_const.CHANNEL_SET_SIDEBAR_DOCS, {
      layout: layout,
      theme: parameters.theme,
      codeTheme: parameters.codeTheme
    });

    return storyComponent;
  };
}

var _default = {
  withReadme: {
    callAsDecorator: withReadmeCallAsDecorator,
    callAsHoc: withReadmeCallAsHoc
  }
};
exports.default = _default;
